### PR TITLE
`sage.rings.{padics,valuation}`: Modularization fixes, `# needs`

### PR DESCRIPTION
**Pull request by mkoeppe**
- Part of: #29705 
- Cherry-picked from: #35095 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/valuation/valuations_catalog.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
...
-from .limit_valuation import LimitValuation
+from sage.misc.lazy_import import lazy_import
+
+lazy_import('sage.rings.padics.padic_valuation', 'pAdicValuation')
+lazy_import('sage.rings.function_field.valuation', 'FunctionFieldValuation')
+lazy_import('sage.rings.valuation.gauss_valuation', 'GaussValuation')
+lazy_import('sage.rings.valuation.trivial_valuation', ['TrivialDiscretePseudoValuation', 'TrivialPseudoValuation', 'TrivialValuation'])
+lazy_import('sage.rings.valuation.limit_valuation', 'LimitValuation')
+
+del lazy_import
```
Comment (dcoudert) [inline src/sage/rings/valuation/valuations_catalog.py:9]: Why is it important to del `lazy_import` here ?

**Review by dcoudert — COMMENTED** [review]
otherwise, LGTM.

File: src/sage/rings/valuation/valuations_catalog.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/rings/valuation/valuations_catalog.py:9]: So that `valuations.<TAB>` does not offer `lazy_import` as one of the completions

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**